### PR TITLE
Resolving Issue 106

### LINKED ISSUE #106: Add primitive for `sklearn.linear_model.LogisticRegression`
https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html

### PR DESCRIPTION
**Pull request by vancoykendall**
Resolve #106 
Created primitive and pipeline for sklearn.linear_model.LogisticRegression

### CODE REVIEW

File: mlprimitives/jsons/sklearn.linear_model.LogisticRegression.json [hunk 1]
```diff
@@ -0,0 +1,148 @@
+{
+    "name": "sklearn.linear_model.LogisticRegression",
+    "contributors": ["Van Coykendall <vcoyk@mit.edu>"],
+    "documentation": "https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html#sklearn.linear_model.LogisticRegression",
```
Comment (Hector-hedb12) [inline mlprimitives/jsons/sklearn.linear_model.LogisticRegression.json]: `#sklearn.linear_model.LogisticRegression` is not necessary.

File: mlprimitives/jsons/sklearn.linear_model.LogisticRegression.json [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                    },
+                    "RandomState":{
+                        "description": "If an array is passed, penalties are assumed to be specific to the targets. Hence they must correspond in number."
+                    },
+                    "NoneType": {
+                        "description": "If none, the random number generator is the RandomState instance used by np.random"
+                    }
+                },
+                "default": null
+            },
```
Comment (Hector-hedb12) [inline mlprimitives/jsons/sklearn.linear_model.LogisticRegression.json]: This is a trailing comma

File: mlprimitives/jsons/sklearn.linear_model.LogisticRegression.json [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                "values": [ "newton-cg", "lbfgs", "liblinear", "sag", "saga" ],
+                "default": "liblinear"
+            },
+            "penalty": {
+                "type": "str",
+                "values": [ "l1", "l2" ],
+                "description": "used to specify the norm used in the penalization. The 'newton-cg', 'sag' and 'lbfgs' solvers support only 'l2' penalties.",
+                "default": "l2"
+            },
+            "c": {
```
Comment (Hector-hedb12) [inline mlprimitives/jsons/sklearn.linear_model.LogisticRegression.json]: parameters are case sensitives, so, I think you meant `C` here

File: mlprimitives/jsons/sklearn.linear_model.LogisticRegression.json [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+            },
+            "multi_class": {
+                "type": "str",
+                "values": [ "ovr", "multinomial", "auto" ],
+                "description": {
+                    "ovr": "If set to 'ovr', then a binary problem is fit for each label.",
+                    "multinomial": "If set to 'multinomial' the loss minimised is the multinomial loss fit across the entire probability distribution. 'multinomial' is unavailable whe 'solver'='liblinear'.",
+                    "auto": "selects 'ovr' if the data is binary, or if 'solver'='liblinear', and otherwise selects 'multinomial'."
+                },
+                "default": "ovr",
```
Comment (Hector-hedb12) [inline mlprimitives/jsons/sklearn.linear_model.LogisticRegression.json]: This is a trailing comma

File: mlprimitives/jsons/sklearn.linear_model.LogisticRegression.json [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+                    }
+                },
+                "description": "These weights will be multiplied with sample_weight (passed through the fit method) if sample_weight is specified.",
+                "default": null
+            },
+            "dual": {
+                "type": "bool",
+                "description": "Dual or primal formulation. Dual formulation is only implemented for l2 penalty with liblinear solver. Prefer dual=False when n_samples > n_features.",
+                "default": false
+            },
```
Comment (Hector-hedb12) [inline mlprimitives/jsons/sklearn.linear_model.LogisticRegression.json]: this is a trailing comma

File: pipelines/sklearn.linear_model.LogisticRegression.json [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{
+	"metadata": {
+		"name": "sklearn.linear_model.LogisticRegression",
+		"data_type": "single_table",
+		"task_type": "classification",
```
Comment (Hector-hedb12) [inline pipelines/sklearn.linear_model.LogisticRegression.json]: this is a trailing comma

File: mlprimitives/jsons/sklearn.linear_model.LogisticRegression.json [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+                "name": "y",
+                "type": "array"
+            }
+        ]
+    },
+    "hyperparameters": {
+    	"fixed": {
+    		"n_jobs": {
+                "type": "int",
+                "description": "Number of CPU cores used when parallelizing over classes if multi_class=’ovr’”. This parameter is ignored when the solver is set to ‘liblinear’ regardless of whether ‘multi_class’ is specified or not. None means 1 unless in a joblib.parallel_backend context. -1 means using all processors",
```
Comment (Hector-hedb12) [inline mlprimitives/jsons/sklearn.linear_model.LogisticRegression.json]: `multi_class=’ovr’”` might be a typo

File: mlprimitives/jsons/sklearn.linear_model.LogisticRegression.json [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+                "default": 0.0001,
+                "range": [ 0.000001, 0.01 ]
+            },
+            "verbose": {
+                "type": "int",
+                "description": "For the liblinear and lbfgs solvers set verbose to any positive number for verbosity.",
+                "default": 0,
+                "range": [ 0, 100 ]
+            },
+            "class_weight": {
```
Comment (csala) [inline mlprimitives/jsons/sklearn.linear_model.LogisticRegression.json]: This should be set as a "fixed" hyperparameter.

File: pipelines/sklearn.linear_model.LogisticRegression.json [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+	},
+	"validation": {
+		"dataset": "boston"
```
Comment (csala) [inline pipelines/sklearn.linear_model.LogisticRegression.json]: Since this is a classification, the dataset used should be `iris` instead of `boston`

**Pull request merged**